packages/gatsby-image/src/__tests__/index.js

```diff
@@ -167,9 +167,39 @@ describe(`<Image />`, () => {
   })
 
   it(`should have the the "critical" prop set "loading='eager'"`, () => {
+    jest.spyOn(global.console, `log`)
+
     const props = { critical: true }
     const imageTag = setup(false, props).querySelector(`picture img`)
     expect(imageTag.getAttribute(`loading`)).toEqual(`eager`)
+    expect(console.log).toBeCalled()
+  })
+
+  it(`should warn if multiple sources with no media are used.`, () => {
```

Comment:
Would this be clearer? The tests are more related to development of `gatsby-image`, where `source` has varied meaning especially regarding the `picture` element, whereas this warning is more to do with the fluid/fixed input array provided to the component.

"should warn if image variants provided are missing media keys."

Reply:
Renamed it as you mentioned